src/workload.rs

```diff
@@ -144,32 +174,108 @@ impl Workload for DoubleInsertion {
                 format!("1*return(\"{}\")", injection),
             )
             .await?;
-            conn.query_drop("BEGIN OPTIMISTIC")?;
+            send!(conn, "BEGIN OPTIMISTIC")?;
             let row = table.new_row();
             let insertion_1 = format!("INSERT INTO {} VALUES ({})", table.name, row.to_string());
             let insertion_2 = format!(
                 "INSERT INTO {} VALUES ({})",
                 table.name,
                 row.next().to_string()
             );
-            println!("execute: {}", insertion_1);
-            let res = conn
-                .query_drop(insertion_1)
-                .and_then(|_| {
-                    println!("execute: {}", insertion_2);
-                    conn.query_drop(insertion_2)
-                })
-                .and_then(|_| {
-                    println!("execute: commit");
-                    conn.query_drop("commit")
-                });
+
+            let res = async {
+                send!(conn, query(insertion_1.as_str()))?;
+                send!(conn, query(insertion_2.as_str()))?;
+                Ok(())
+            }
+            .await;
+
+            if res.is_err() {
+                send!(conn, "ROLLBACK")?;
+            }
+            info!(log, "{:?}", res);
+
+            collect_result(res, results, &table, injection, pool.clone()).await;
+            disable_failpoint(
+                client,
+                "github.com/pingcap/tidb/table/tables/corruptMutations",
+            )
+            .await?;
+        }
+
+        Ok(())
+    }
+}
+
+// 2 txns. The first writes corrupted data and the second reads it. Check if Assertion can detect it.
```

Comment:
The second txn should be a write too (and your code is write...) Assertion is expected to take effect when trying to write correct data on existing corrupted data or write corrupted data on existing correct data.

Reply:
Right